src/test/java/usecasetesting/ReplyUseCaseTest.java

```diff
@@ -0,0 +1,162 @@
+        questionGateway.saveQuestion(question);
+
+        Question closedQuestion = new Question();
+        closedQuestion.setQuestionId(CLOSED_QUESTION_ID);
+        questionGateway.saveQuestion(closedQuestion);
+
+        Client client = new Client();
+        client.setUserId(CLIENT_ID);
+        client.addQuestion(question);
+        clientGateway.addUser(client);
+
+        Attorney attorney = new Attorney();
+        attorney.setUserId(ATTORNEY_ID);
+        attorneyGateway.addUser(attorney);
+
+        Attorney secondAttorney = new Attorney();
+        secondAttorney.setUserId(SECOND_ATTORNEY_ID);
+        attorneyGateway.addUser(secondAttorney);
+    }
+
+    @Test
+    public void testClientReply(){
+        setUpReplyUseCase();
+        PostRequestModel inputData1 = new PostRequestModel(QUESTION_ID, CLIENT_ID, "Test text");
+        postInputBoundary.createPost(inputData1);// null pointer error caused by posts not initialized by Question
+        Question question = questionGateway.getQuestion(QUESTION_ID);
+        Post post1 = question.getPosts().get(0);
+        assertEquals(post1.getBelongsTo(), CLIENT_ID);
+        assertEquals(post1.getPostText(), "Test text");
+    }
+
+    @Test
+    public void testAttorneyFirstReply(){
+        setUpReplyUseCase();
+
+        PostRequestModel inputData2 = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData2);
+        Question question = questionGateway.getQuestion(QUESTION_ID);
+        Post post2 = question.getPosts().get(0);
+        assertEquals(post2.getBelongsTo(), ATTORNEY_ID);
+        assertEquals(post2.getPostText(), "Test text");
+        User user = attorneyGateway.getUser(ATTORNEY_ID);
+        Question attorneyQuestion = user.getQuestionsList().get(0);// AttorneyQuestionList is not updated
+        assertEquals(attorneyQuestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyQuestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyQuestion.isTaken(), true);
+    }
+    @Test
+    public void testAttorneyFollowUp(){
+        setUpReplyUseCase();
+        PostRequestModel inputData = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData);
+
+        User user = attorneyGateway.getUser(ATTORNEY_ID);
+        Question attorneyquestion = user.getQuestionsList().get(0);// AttorneyQuestionList is not updated
+        assertEquals(attorneyquestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyquestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyquestion.isTaken(), true);
+
+        PostRequestModel inputData2 = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text2");// Attorney
+        postInputBoundary.createPost(inputData2);
+        Question question = questionGateway.getQuestion(QUESTION_ID);
+        Post post2 = question.getPosts().get(1);
+        assertEquals(post2.getBelongsTo(), ATTORNEY_ID);
```

Comment:
Could be nice to include a message for the assertions to make sure we know where the failure occurrs if there is one.